internal/daemon/controller/handler.go

```diff
@@ -57,18 +57,36 @@ type HandlerProperties struct {
 	CancelCtx      context.Context
 }
 
+func isUiRequest(req *http.Request) bool {
+	return !strings.HasPrefix(req.URL.Path, "/v1/")
+}
+
+// createMuxWithEndpoints performs all response logic for boundary, using isUiRequest
+// for unified logic between responses and headers.
+func createMuxWithEndpoints(c *Controller, props HandlerProperties) (http.Handler, error) {
+	grpcGwMux := newGrpcGatewayMux()
+	if err := registerGrpcGatewayEndpoints(props.CancelCtx, grpcGwMux, gatewayDialOptions(c.apiGrpcServerListener)...); err != nil {
+		return nil, err
+	}
+
+	uiHandler := handleUi(c)
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if isUiRequest(req) {
+			uiHandler.ServeHTTP(w, req)
+		} else {
+			grpcGwMux.ServeHTTP(w, req)
+		}
+	}), nil
+}
+
```

Comment:
I don't follow this change that moves away from calling `mux.Handle` and instead adds this `isUiRequest` method.

Reply:
After digging into https://github.com/hashicorp/go-secure-stdlib/pull/49 I think I see what is going on here. It still seems odd to introduce this `!strings.HasPrefix` check with `isUiRequest` instead of leveraging the `mux.Handle` and allowing the mux to do more efficient matching, but I am not seeing a way to do that given the implementation in listenerutil.

Reply:
[latest commit](https://github.com/hashicorp/boundary/pull/2587/commits/a5263b3ec52e4f4be733bc3b58f677fb7a209ede) swaps `strings.HasPrefix` for [`mux.Handler`](https://pkg.go.dev/net/http?utm_source=gopls#ServeMux.Handler), which is used internally by `mux.ServeHTTP`, so we can go back to using mux's logic for serving as well